packages/block-library/src/search/edit.js

```diff
@@ -130,12 +134,32 @@ export default function SearchEdit( {
 	const isButtonPositionOutside = 'button-outside' === buttonPosition;
 	const hasNoButton = 'no-button' === buttonPosition;
 	const hasOnlyButton = 'button-only' === buttonPosition;
+	const searchFieldRef = useRef();
+	const buttonRef = useRef();
 
 	const units = useCustomUnits( {
 		availableUnits: [ '%', 'px' ],
 		defaultValues: { '%': PC_WIDTH_DEFAULT, px: PX_WIDTH_DEFAULT },
 	} );
 
+	useEffect( () => {
+		if ( hasOnlyButton && ! isSelected ) {
+			setAttributes( {
+				isSearchFieldHidden: true,
+			} );
+		}
+	}, [ isSelected ] );
```

Comment:
Ideally we should list all the dependencies for this effect
```suggestion
	}, [ hasOnlyButton, isSelected, setAttributes ] );
```

Reply:
[0cb5ef2](https://github.com/WordPress/gutenberg/pull/50487/commits/0cb5ef222e370ea31f0369abf46ad5067449af69)